pkg/mdformatter/linktransformer/validator.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Bartłomiej Płotka @bwplotka
+// Licensed under the Apache License 2.0.
+
+package linktransformer
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	gitHubAPIURL = "https://api.github.com/repos/%v/%v?sort=created&direction=desc&per_page=1"
+)
+
+type GitHubResponse struct {
+	Number int `json:"number"`
+}
+
+// Match link with any one of provided validators.
+func CheckValidators(dest string, v Config) bool {
+	for _, val := range v.Validate.Validators {
+		if val._regex.MatchString(dest) {
+			if val.Type == "github" {
+				// Find rightmost index of match i.e, where regex match ends.
+				// This will be where issue/PR number starts. Split incase of section link and convert to int.
+				idx := val._regex.FindStringIndex(dest)
+				stringNum := strings.Split(dest[idx[1]:], "#")
+				num, err := strconv.Atoi(stringNum[0])
+				// If number in link does not exceed then link is valid. Otherwise will be checked by v.c.Visit.
+				if val._maxnum >= num && err == nil {
+					return true
+				}
+				return false
+			}
+			return true
+		}
+	}
+	return false
+}
+
+// If type is "github", change regex and add maxnum.
+func CheckGitHub(v Config) error {
+	for i := range v.Validate.Validators {
+		if v.Validate.Validators[i].Type == "github" {
+			regex, maxnum, err := getGitHubRegex(v.Validate.Validators[i].Regex)
+			if err != nil {
+				return err
+			}
+			v.Validate.Validators[i]._regex = regex
+			v.Validate.Validators[i]._maxnum = maxnum
+		}
+	}
+	return nil
+}
+
+// Get GitHub pulls/issues regex from repo name.
+func getGitHubRegex(repoRe string) (*regexp.Regexp, int, error) {
+	if repoRe != "" {
+		// Get reponame from regex.
+		idx := strings.Index(repoRe, `\`)
```

Comment:
Hm, it is bit confusing. What about either expecting named regex group? `(?P<repo>.*)

Reply:
I'll try to implement this!